robot_ws/src/voice_interaction_robot/src/voice_interaction_robot/voice_input.py

```diff
@@ -132,23 +132,21 @@ def get_record_device_index(self):
             your machines default device
         """
 
-        rospy.loginfo("Attempting to find device named 'record'")
+        self.get_logger().info("Attempting to find device named 'record'")
         for i in range(self.p.get_device_count()):
             device = self.p.get_device_info_by_index(i)
             if device['name'] == 'record':
-                rospy.loginfo("Found device 'record' at index %d" % i)
+                self.get_logger().info("Found device 'record' at index %d" % i)
                 self.input_device_index = i
                 return
-        rospy.loginfo("Could not find device named 'record', falling back to default recording device")
+        self.get_logger().info("Could not find device named 'record', falling back to default recording device")
```

Comment:
Don't need to change since this is the same in ROS1, but this feels like it could be made into a more generic function like `get_device_from_index(device_name)` or something that we call with `device_name="record"` or at least set a variable `device_name=record` at the start instead of using the string `record` all over the place.

Reply:
Good idea. Done.